lib/spack/spack/environment.py

```diff
@@ -605,7 +605,7 @@ def remove(self, query_spec, force=False):
         # try abstract specs first
         matches = []
         if not query_spec.concrete:
-            matches = [s for s in self.user_specs if s.satisfies(query_spec)]
+            matches = [s for s in self.user_specs if s == query_spec]
```

Comment:
This line should not be merged as-is. This will require users to specify the full spec they wish to remove, rather than only enough to make it unique.